File: include/apl/errors/LengthError.java
package APL.errors;

import APL.types.*;

public class LengthError extends APLError {
  // public LengthError(std::string s) {
  //   super(s);
  // }
  public LengthError(std::string s, Tokenable fun) {
    super(s, fun);
  }
  public LengthError(std::string s, Callable fun, Tokenable cause) {
    super(s, fun, cause);
  }
}

File: APL/errors/LengthError.cpp
#include <APL/LengthError.h>

namespace APL::errors
{
    // using namespace APL::types;

    LengthError::LengthError(std::string const& s, std::shared_ptr<APL::types::Tokenable> fun) : APLError(s, fun)
    {
    }

    LengthError::LengthError(std::string const& s, std::shared_ptr<APL::types::Callable> fun, std::shared_ptr<APL::types::Tokenable> cause) : APLError(s, fun, cause)
    {
    }
}
